packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -240,6 +240,19 @@ export const usePickerValue = <
     };
   });
 
+  React.useEffect(() => {
```

Comment:
We should avoid `setState` in `useEffect` since they cause double render.
Instead we should do the state update in the render (see https://swizec.com/blog/react-can-update-state-during-render/)

Reply:
Thanks for bringing this problem up. 🙏 
Do you have something like this (https://github.com/mui/mui-x/pull/15435/commits/54b8d751b920df49038f5236ae15057398af62f3) in mind?

Reply:
We can simplify the condition, `getTimezone` only returns null if the value is null or invalid so I think we don't need to check both for the existence of `timezoneFromDraftValue` and `dateState.draft`.

But other than that yes that was my idea :ok_hand: 
We clearly have other `setState` in `useEffect` in our codebase that we could remove to improve performances in the future.

If `getTimezone` was heavy we could memoize `timezoneFromDraftValue` but I don't think it's worth it here.

```ts
  const [prevTimezoneProp, setPrevTimezoneProp] = React.useState(timezoneProp);
  const timezoneFromDraftValue = valueManager.getTimezone(utils, dateState.draft);

  if (timezoneProp !== prevTimezoneProp)  {
    setPrevTimezoneProp(timezoneProp);

    if (timezoneProp && timezoneFromDraftValue !== timezoneProp) {
      setDateState((prev) => ({
        ...prev,
        draft: valueManager.setTimezone(utils, timezoneProp, prev.draft),
      }));
    }
  }
```

Reply:
Simplified the solution. 👌 
I'm just curious, should we fire onChange when we internally change the value timezone due to an update of `timezone`? 🤔 🤷